### PR TITLE
Fix watch attempting to reuse ports that are not yet released by the old process

### PR DESCRIPTION
**Pull request by gvilums**
### What does this PR do?
This PR changes how `--watch` mode reloads work on windows. Previously, we would have one top-level process wait on a job object containing the actual bun process, and exit once all children have exited. The child process would be the one actually spawning a copy and terminating itself.

Now, if the child process wants to reload itself, it exits with a magic exit code that is picked up by the top-level process, which in turn restarts the child.

With the new setup, the process is only restarted once the previous copy has exited. This should fix issues where ports were still in use by the old version of a process when the new copy was spawned.

### How did you verify your code works?
Manual testing.

### CODE REVIEW

File: src/__global.zig [hunk 1]
```diff
@@ -92,10 +92,14 @@ pub fn runExitCallbacks() void {
...
 pub fn exit(code: u8) noreturn {
+    exitWide(@as(u32, code));
+}
+
+pub fn exitWide(code: u32) noreturn {
     runExitCallbacks();
     Output.flush();
     std.mem.doNotOptimizeAway(&Bun__atexit);
-    std.c.exit(code);
+    std.c.exit(@bitCast(code));
```
Comment (Jarred-Sumner) [inline src/__global.zig:102]: This will fail to compile on Linux, no?

Comment (Jarred-Sumner) [reply src/__global.zig:102]: I think you want `@intCast`

Comment (gvilums) [reply src/__global.zig:102]: std.c.exit takes a `c_int` on all platforms I think

Comment (gvilums) [reply src/__global.zig:102]: or wait, maybe not

**Pull request merged**